backend/amd64/cfg_stack_operands.ml

```diff
@@ -78,54 +84,67 @@ let binary_operation
     | Result_cannot_be_on_stack ->
       check_lengths instr ~of_arg:2 ~of_res:1
   end;
-  begin match is_spilled instr.arg.(0), is_spilled instr.arg.(1) with
-  | false, false ->
-    begin match result with
-    | No_result | Result_can_be_on_stack ->
-      All_spilled_registers_rewritten
-    | Result_cannot_be_on_stack ->
-      May_still_have_spilled_registers
-    end
-  | false, true ->
-    use_stack_operand map instr.arg 1;
-    begin match result with
-    | No_result ->
-      All_spilled_registers_rewritten
-    | Result_can_be_on_stack | Result_cannot_be_on_stack ->
-      May_still_have_spilled_registers
-    end
-  | true, false ->
-    (* note: slightly different from the case above, because arg.(0) and res.(0) are the same. *)
-    begin match result with
-    | No_result ->
-      use_stack_operand map instr.arg 0;
-      All_spilled_registers_rewritten
-    | Result_can_be_on_stack ->
-      use_stack_operand map instr.res 0;
-      use_stack_operand map instr.arg 0;
-      All_spilled_registers_rewritten
-    | Result_cannot_be_on_stack ->
-      use_stack_operand map instr.arg 0;
-      May_still_have_spilled_registers
-    end;
-  | true, true ->
-    if Reg.same instr.arg.(0) instr.arg.(1) then
-      May_still_have_spilled_registers
-    else begin
-      match result with
+  let already_has_memory_operand =
+    is_stack_operand instr.arg.(0)
+    || is_stack_operand instr.arg.(1)
+    || (match result with
+      | No_result -> false
+      | Result_cannot_be_on_stack ->
+        assert (not (is_stack_operand instr.res.(0)));
+        false
+      | Result_can_be_on_stack -> is_stack_operand instr.res.(0))
```

Comment:
This case is implied by L88 `is_stack_operand instr.arg.(0)` and L83 res.(0)==arg.(0).

Reply:
For the avoidance of doubt, your point is that
line 95 is actually unreachable, right?

Reply:
yes, under short circuiting  of || .